cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -998,11 +1003,26 @@ def _validate_obsm(self):
                 # Skip over the subsequent checks that require the value to be an array
                 continue
 
-            if len(value.shape) < 2 or value.shape[0] != self.adata.n_obs or value.shape[1] < 2:
-                issue_list.append(
-                    f"All embeddings must have as many rows as cells, and at least two columns."
-                    f" 'adata.obsm['{key}']' has shape of '{value.shape}'."
+            if len(value.shape) < 2:
+                self.errors.append(
+                    f"All embeddings must at least two dimensions. 'adata.obsm['{key}']' has a shape length of '{len(value.shape)}'."
                 )
+            else:
+                if value.shape[0] != self.adata.n_obs:
+                    self.errors.append(
+                        f"All embeddings must have as many rows as cells. 'adata.obsm['{key}']' has rows='{value.shape[0]}'."
+                    )
+
+                if unknown_key and value.shape[1] < 1:
+                    self.errors.append(
+                        f"All other embeddings must have at least one column. 'adata.obsm['{key}']' has columns='{value.shape[1]}'."
```

Comment:
nit: in this context, I don't think the curator reading would know what "other embeddings" mean. Maybe "any embeddings not specified in the schema reference"?